core/store/migrations/0034_external_job_id.go

```diff
@@ -0,0 +1,33 @@
+package migrations
+
+import "gorm.io/gorm"
+
+const (
+	up34 = `
+               CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+               ALTER TABLE direct_request_specs DROP COLUMN on_chain_job_spec_id;
+               ALTER TABLE webhook_specs DROP COLUMN on_chain_job_spec_id;
+               ALTER TABLE jobs ADD COLUMN external_job_id uuid NOT NULL DEFAULT uuid_generate_v4();
```

Comment:
```suggestion
ALTER TABLE jobs 
ADD COLUMN external_job_id uuid NOT NULL DEFAULT uuid_generate_v4(),           
ADD CONSTRAINT external_job_id_uniq UNIQUE(external_job_id),
ADD CONSTRAINT non_zero_uuid_check CHECK (external_job_id &lt;&gt; '00000000-0000-0000-0000-000000000000');
```

These can be combined to